org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -466,7 +463,8 @@ public static IJavaElement findElementAtSelection(ITypeRoot unit, int line, int
 		}
 		if (unit instanceof IClassFile) {
 			IClassFile classFile = (IClassFile) unit;
-			String contents = disassemble(classFile);
+			ContentProviderManager contentProvider = new ContentProviderManager(preferenceManager);

```

Comment:
We want to keep a single instance of ContentProviderManager across the JVM, or else it's cache of decompiled classes will have no value